src/crystal/system/win32/env.cr

```diff
@@ -83,12 +87,18 @@ module Crystal::System::Env
     # `System.to_wstr` here
     String.build do |io|
       env.each do |(key, value)|
-        if key.includes?('=') || key.empty?
-          raise ArgumentError.new("Invalid env key #{key.inspect}")
-        end
+        check_valid_key(key)
         io << key.check_no_null_byte("key") << '=' << value.check_no_null_byte("value") << '\0'
       end
       io << '\0'
     end.to_utf16.to_unsafe
   end
+
+  private def self.check_valid_key(key : String, &)
+    yield if key.empty? || key.includes?('=')
+  end
+
+  private def self.check_valid_key(key : String)
+    check_valid_key(key) { raise ArgumentError.new("Invalid env key #{key.inspect}") }
+  end
```

Comment:
This might be a bit stylistic and subjective, but I think it's also quite relevant for code readability:
Why not have methods `valid_key?` and `check_valid_key`? Then non-raising resolutions would be something like `return false unless valid_key?(key)` which is IMO more easily understandable than `check_valid_key(key) { return false }`.